src/node.rs

```diff
@@ -87,255 +53,232 @@ impl Default for Taffy {
+    pub fn child_at_index(&self, parent: Node, child_index: usize) -> TaffyResult<Node> {
+        let child_count = self.children[parent].len();
         if child_index >= child_count {
-            return Err(error::InvalidChild::ChildIndexOutOfBounds { parent, child_index, child_count });
+            return Err(error::TaffyError::ChildIndexOutOfBounds { parent, child_index, child_count });
         }
 
-        Ok(self.ids_to_nodes[&self.forest.children[id][child_index]])
+        Ok(self.children[parent][child_index])
     }
 
     /// Returns the number of children of the `parent` [`Node`]
-    pub fn child_count(&self, parent: Node) -> Result<usize, error::InvalidNode> {
-        let id = self.find_node(parent)?;
-        Ok(self.forest.children[id].len())
+    pub fn child_count(&self, parent: Node) -> TaffyResult<usize> {
+        Ok(self.children[parent].len())
     }
 
     /// Returns a list of children that belong to the [`Parent`]
-    pub fn children(&self, parent: Node) -> Result<Vec<Node>, error::InvalidNode> {
-        let id = self.find_node(parent)?;
-        Ok(self.forest.children[id].iter().map(|child| self.ids_to_nodes[child]).collect())
+    pub fn children(&self, parent: Node) -> TaffyResult<Vec<Node>> {
+        Ok(self.children[parent].iter().copied().collect::<_>())
     }
 
     /// Sets the [`Style`] of the provided `node`
-    pub fn set_style(&mut self, node: Node, style: FlexboxLayout) -> Result<(), error::InvalidNode> {
-        let id = self.find_node(node)?;
-        self.forest.nodes[id].style = style;
-        self.forest.mark_dirty(id);
+    pub fn set_style(&mut self, node: Node, style: FlexboxLayout) -> TaffyResult<()> {
+        self.nodes[node].style = style;
+        self.mark_dirty(node)?;
         Ok(())
     }
 
     /// Gets the [`Style`] of the provided `node`
-    pub fn style(&self, node: Node) -> Result<&FlexboxLayout, error::InvalidNode> {
-        let id = self.find_node(node)?;
-        Ok(&self.forest.nodes[id].style)
+    pub fn style(&self, node: Node) -> TaffyResult<&FlexboxLayout> {
+        Ok(&self.nodes[node].style)
     }
 
     /// Return this node layout relative to its parent
-    pub fn layout(&self, node: Node) -> Result<&Layout, error::InvalidNode> {
-        let id = self.find_node(node)?;
-        Ok(&self.forest.nodes[id].layout)
+    pub fn layout(&self, node: Node) -> TaffyResult<&Layout> {
+        Ok(&self.nodes[node].layout)
     }
 
     /// Marks the layout computation of this node and its children as outdated
-    pub fn mark_dirty(&mut self, node: Node) -> Result<(), error::InvalidNode> {
-        let id = self.find_node(node)?;
-        self.forest.mark_dirty(id);
+    pub fn mark_dirty(&mut self, node: Node) -> TaffyResult<()> {
+        /// Performs a recursive depth-first search up the tree until the root node is reached
+        ///
+        ///  WARNING: this will stack-overflow if the tree contains a cycle
```

Comment:
Should this comment be pulled up into the function comment so it's visible for users of the API?